### PR TITLE
Clear meta data when unloading all records for a type. Fixes #2772

### LINKED ISSUE #2772: store.unloadAll does not reset meta data
Using ember-cli, given type `conversation-preview`, after calling `this.get('store').unloadAll('conversationPreview')` a call to `this.get('store').metadataFor('conversation-preview');` returns the stale meta data.

### PR DESCRIPTION
**Pull request by runspired**

### CODE REVIEW

File: packages/ember-data/lib/system/store.js [hunk 1]
```diff
@@ -1073,6 +1073,7 @@ Store = Service.extend({
     }
 
     typeMap.findAllCache = null;
+    this.typeMapFor(type).metadata = {};
```
Comment (wecc) [inline packages/ember-data/lib/system/store.js]: Probably want to do `Ember.create(null)` instead of `{}` here

Comment (runspired) [inline packages/ember-data/lib/system/store.js]: Changed.  Are there docs somewhere on Ember.create ? It's not in the website api docs. I'm assuming this is the underlying instantiation method (similar to Ember.set or Ember.get) that get's added to Ember.Object?

File: packages/ember-data/lib/system/store.js [hunk 2]
```diff
@@ -1073,6 +1073,7 @@ Store = Service.extend({
     }
 
     typeMap.findAllCache = null;
+    this.typeMapFor(type).metadata = Ember.create(null);
```
Comment (wecc) [inline packages/ember-data/lib/system/store.js]: You already have `typeMap` available here, another call to `typeMapFor()` should not be needed.

**Comment by wecc** [comment]
Had a brief chat with @bmac about maybe clearing the typeMap for the specified type all together instead of nullifying separate properties (`findAllCache`, `metadata`). Thoughts @igorT?

**Comment by runspired** [comment]
@wecc I personally think that makes sense.  I'm assuming if desired doing so would involve doing this after unloading the records?

```
var typeMaps = get(this, 'typeMaps');
var guid = Ember.guidFor(type);
if (typeMaps[guid]) {
    delete typeMaps[guid];
}
```

**Comment by runspired** [comment]
@wecc also, any idea why the build can't complete? Can't get the tests to run locally, just hangs forever, seems same issue is happening with the CI tests.

**Comment by wecc** [comment]
@runspired failing build should be fixed on master, can you rebase?

**Comment by igorT** [comment]
Thanks for the PR! Can you squash/rebase and I'll or @wecc will merge

**Comment by runspired** [comment]
@igorT @wecc squashed and rebased

**Pull request merged**